### PR TITLE
[bug 1225987] Add Opt Out to Mobile Widget Snippet

### PR DESCRIPTION
**Pull request by ckprice**
@glogiotatidis r?

### CODE REVIEW

**Comment by glogiotatidis** [comment]
@schalkneethling do you have time to r?

File: campaigns/mobile/mobile-snippet-widget.html [hunk 1]
```diff
@@ -226,6 +226,49 @@
...
+      }
+      #firefox-mobile-activation-{{ snippet_id }}:hover #xbutton-{{ snippet_id }} {
+         opacity: 1;
+      }
+      #xbutton-{{ snippet_id }} + #block-snippet-overlay-{{ snippet_id }} {
+        opacity: 0;
+        transition: opacity 100ms linear;
```
Comment (schalkneethling) [inline campaigns/mobile/mobile-snippet-widget.html]: You need to use seconds so:
`transition: opacity .1s linear;`

File: campaigns/mobile/mobile-snippet-widget.html [hunk 2]
```diff
@@ -226,6 +226,49 @@
...
+      }
+      #xbutton-{{ snippet_id }}:hover + #block-snippet-overlay-{{ snippet_id }} {
+        opacity: 0.03;
+        background-color: #000000;
+        box-shadow: 0pt 0pt 0pt white;
```
Comment (schalkneethling) [inline campaigns/mobile/mobile-snippet-widget.html]: No need for units for 0 based values.

File: campaigns/mobile/mobile-snippet-widget.html [hunk 3]
```diff
@@ -226,6 +226,49 @@
...
+      #firefox-mobile-activation-{{ snippet_id }}:hover #xbutton-{{ snippet_id }} {
+         opacity: 1;
+      }
+      #xbutton-{{ snippet_id }} + #block-snippet-overlay-{{ snippet_id }} {
+        opacity: 0;
+        transition: opacity 100ms linear;
+      }
+      #xbutton-{{ snippet_id }}:hover + #block-snippet-overlay-{{ snippet_id }} {
+        opacity: 0.03;
+        background-color: #000000;
```
Comment (schalkneethling) [inline campaigns/mobile/mobile-snippet-widget.html]: Nit: Could be just `background-color: #000;`

File: campaigns/mobile/mobile-snippet-widget.html [hunk 4]
```diff
@@ -226,6 +226,49 @@
...
+      }
+      #firefox-mobile-activation-{{ snippet_id }}:hover #xbutton-{{ snippet_id }} {
+         opacity: 1;
+      }
+      #xbutton-{{ snippet_id }} + #block-snippet-overlay-{{ snippet_id }} {
+        opacity: 0;
+        transition: opacity 100ms linear;
+      }
+      #xbutton-{{ snippet_id }}:hover + #block-snippet-overlay-{{ snippet_id }} {
+        opacity: 0.03;
```
Comment (schalkneethling) [inline campaigns/mobile/mobile-snippet-widget.html]: Is there really a perceivable difference between `0` and `0.03`?

File: campaigns/mobile/mobile-snippet-widget.html [hunk 5]
```diff
@@ -226,6 +226,49 @@
...
+        background-color: #000000;
+        box-shadow: 0pt 0pt 0pt white;
+        border: solid 0pt white;
```
Comment (schalkneethling) [inline campaigns/mobile/mobile-snippet-widget.html]: Same here

File: campaigns/mobile/mobile-snippet-widget.html [hunk 6]
```diff
@@ -226,6 +226,49 @@
...
+         right: 0;
+         top: 0;
+         z-index: 10;
+         opacity: 0;
+         height: 20px;
+         width: 20px;
+         background-color: rgba(0,0,0,0);
+         background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAMAAAAp4XiDAAAAeFBMVEUAAAAAAAAAAAABAQEAAAAAAAAAAAAAAAAAAAAAAADDw8OQkJAAAAAAAAAAAAAAAAD9/f38/Pz6+vr5+fnt7e3i4uLd3d3Z2dmgoKBdXV0mJibv7++amprNzc2amprv7+////9zc3OwsLD29vaPj4+xsbHCwsLBwcHrzmRyAAAAIHRSTlMABAg9HhEMNxkir38xLCkV/Pr39OHTzceOYUrliLuJ5ik+KlgAAAGhSURBVEjHnZbrdoIwDIAXWii3UuV+0wV0+v5vOEBH2ajg8v3y6PlO2pgm+TAAYD0B+HgDsBxbuXzCVbZj7VgAju16USjYhAgjz7WdrVjgKC5FHfh5kxwOSZP7QS0kVw68PJLNJQuKIy44FgGT3LbAHMKNWJviirRlkWsKNIQIqwyNZFU4BFobnihjfEFcCm92tMF83MBnfxwYjBNuchocWN6cCx938AVf5MBywxJ3KUPX0seKqnhfiavIhvlYLMM3yNjP0UDJFmcuN/zF7aI/t1I9lCFIqo1r1y+Nr+6qnXQI87iJDJZGNzra6JZOIG2Y0iWK+bt7N9JrY+Q+/1yIKWmOVy9qt3862lgGPdaeM2U4QDQ52tAEY55BhT4aHS1r/FDBeJUcTY7JwHy8jMVZgybHZGDD+KQkuHbMBiZP5YBrx2zggaCQDka5PiHJhL9yq2B6c8FslmVvKsud4u8Nxb/3xPr1E9t9yP3qIf+vXZylAkJTIrQ+QoMltHHCsCCMJMLgo49XPcTPpiF+1kOcsCrsLSSfeiGhrj305Yqwwn0Dwi/I5rQKpwMAAAAASUVORK5CYII=");
+         background-repeat: no-repeat;
+         background-size: 20px 20px;
```
Comment (schalkneethling) [inline campaigns/mobile/mobile-snippet-widget.html]: I would add `margin: 10px;` here as well.

**Comment by schalkneethling** [comment]
I guess clicking the `x` not actually closing/removing the send to device widget is expected in preview mode?

**Comment by ckprice** [comment]
@schalkneethling have you downloaded this https://github.com/Osmose/snippet-switcher?

It's an add-on you can use to test a snippet on about:home which is more reliable than preview.

Basically you add a "Name" CMR (on DEV, I just add 'ui-tour-test' every time) in the snippet configuration, then in the add-on, change your server to https://snippets.allizom.org/ and update the 'Name' field. Example: http://cl.ly/image/0z0R162N3P1Z

**Comment by schalkneethling** [comment]
Yeah, I have tried it and could not get much joy out of it. I will give it a try again.

**Comment by ckprice** [comment]
@schalkneethling updated!

**Comment by schalkneethling** [comment]
r+ @glogiotatidis merge at will. Oooh, unless we need @ckprice to squash first.

**Comment by glogiotatidis** [comment]
Squashed and merged, thanks folks!

https://github.com/mozilla/snippets/commit/0b523a696ee9ebfd6b70ee95ebe6cdd400e1761d

Also updated on production

**Pull request closed without merging**